decidim-core/app/models/decidim/participatory_process_user_role.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+module Decidim
+  # Defines a relation between a user and a participatory process, and what
+  # kind of relation does the user has.
+  class ParticipatoryProcessUserRole < ApplicationRecord
+    belongs_to :user, foreign_key: "decidim_user_id", class_name: Decidim::User
+    belongs_to :participatory_process, foreign_key: "decidim_participatory_process_id", class_name: Decidim::ParticipatoryProcess
+
+    ROLES = %w(admin).freeze
+    validates :role, inclusion: { in: ROLES }
```

Comment:
Instead of creating a full-fledged query object to solve that `concern` issue, you could always add a class method like:

```ruby
def self.participatory_processes_for(user)
  # Whatever
end

def self.users_for(participatory_process)
  # Whatever
end
```